### ISSUE TITLE
Input: Should revert to the last valid value when Input component loses focus

### ISSUE THREAD

**Issue by anranruye** [ant-design-blazor/ant-design-blazor#1553]
### Describe the bug

When use `Input<NotStringType>`, the component input value should revert to the last valid value when Input component loses focus if the current input value is not valid.

![image](https://user-images.githubusercontent.com/54608128/118619218-f67a9a80-b7f6-11eb-88cb-d7afe83d7f9f.png)

### Steps to reproduce (please include code)

https://ant-design-blazor.gitee.io/en-US/components/input#components-input-demo-Generic
### Further technical details
- AntDesign Nuget Package version
- Include the output of `dotnet --info`
- The IDE (VS / VS Code/ VS4Mac) you're running on, and it's version

**Comment by anddrzejb** [comment]
I could probably include that as a part to my PR #1530, unless you @anranruye are already working on this? However, I am away now, so it may take me a week or 2 to finish this up.
@ElderJames maybe we could have a label like `up for grabs` to have explicit indication that a particular issue is not being worked on but we consider it valuable/important and it is probably a simple fix? This may seem similar to `help wanted`, but `help wanted` is really broad, while `up for grabs` could be a good issue for someone considering one of their first contribution.

**Comment by anranruye** [comment]
@anddrzejb  I'm not working on this. It's good to hear that you prepare to solve this. 

Do you have any ideas about a similar issue - reverting when users input an invalid character or paste from clipboard?

**Comment by anddrzejb** [comment]
I guess this could be controlled in the `OnInput` event handler. If we are going to allow this, there probably should be a `[Parameter]` toggling such functionality (for example `ValidateTypeOnInput` or smth like that).

**Comment by anddrzejb** [comment]
@anranruye I had a closer look at this and it proved to be a bit more involved. I have a working solution for `OnBlur`, but my change is substantial and requires extensive testing (especially that it also affects other components that rely on `Input`). I think your other suggestion is also achievable, though not for all data types. The issue is with `DateTime` or `Guid`. Since one cannot know if data entered is correct until whole value is entered, a character-per-character validation is not possible (same with partial copy-paste). Unless we implement some kind of pattern-matching. That is much more work I think. Maybe we could use here regex.

**Comment by anranruye** [comment]
@anddrzejb Yes, here regex is very useful. Since we have decided to support localization for input components, there may be much more work.
